src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,105 @@
+import './NewCommentForm.scss';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../../types';
+
+type Props = {
+  commentsList: Comment[] | undefined;
+  setCommentsList: (arg:Comment[]) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+                                                  commentsList, setCommentsList,
+                                                }) => {
+  const [yourname, setYourname] = useState('');
+  const [youremail, setYouremail] = useState('');
+  const [yourcomment, setYourcomment] = useState('');
+  const [error, setError] = useState(false);
+
+  const addComment = (newComment: Comment) => {
+    if (commentsList) {
+      commentsList.unshift(newComment);
+      const copy = [...commentsList];
+
+      setCommentsList(copy);
+    }
```

Comment:
Since useState is not updated immediately, it is best to use the setState function to update the state based on the previous value. 
For example: 
`setCommentsList(prevCommentList => [ newComment, ...prevCommentList ])`